app/zone-details/index.tsx

```diff
@@ -0,0 +1,61 @@
+import { useLocalSearchParams } from 'expo-router'
+import { View } from 'react-native'
+
+import SegmentBadge from '@/components/info/SegmentBadge'
+import Divider from '@/components/shared/Divider'
+import FlexRow from '@/components/shared/FlexRow'
+import Icon from '@/components/shared/Icon'
+import ScreenView from '@/components/shared/ScreenView'
+import Typography from '@/components/shared/Typography'
+import { useTranslation } from '@/hooks/useTranslation'
+import { useMapZone } from '@/state/hooks/useMapZone'
+
+type ZoneDetailsParamas = {
+  id: string
+}
+
+const ZoneDetailsScreen = () => {
+  // reason for not passing the whole object: https://reactnavigation.org/docs/params/#what-should-be-in-params
+  const zoneDetailsParams = useLocalSearchParams<ZoneDetailsParamas>()
+  const t = useTranslation()
+
+  const id = zoneDetailsParams.id ? Number.parseInt(zoneDetailsParams.id, 10) : -1
+
+  const zone = useMapZone(id, true)
+
+  if (!zone) {
```

Comment:
I know that it is a negative condition, but it saves indentation and makes it clearer what happens, for return value of Elements, I would do it this way